vortex-datetime-dtype/src/temporal.rs

```diff
@@ -43,6 +65,36 @@ impl TemporalMetadata {
             None
         }
     }
+
+    pub fn to_jiff(&self, v: i64) -> VortexResult<TemporalJiff> {
+        match self {
+            TemporalMetadata::Time(TimeUnit::D) => {
+                vortex_bail!(InvalidArgument: "Invalid TimeUnit TimeUnit::D for TemporalMetadata::Time")
+            }
+            TemporalMetadata::Time(unit) => Ok(TemporalJiff::Time(
+                Time::MIN.checked_add(unit.to_jiff_span(v))?,
+            )),
+            TemporalMetadata::Date(unit) => match unit {
+                TimeUnit::D | TimeUnit::Ms => Ok(TemporalJiff::Date(
+                    Date::new(1970, 1, 1)?.checked_add(unit.to_jiff_span(v))?,
+                )),
+                _ => {
+                    vortex_bail!(InvalidArgument: "Invalid TimeUnit {} for TemporalMetadata::Time", unit)
+                }
+            },
+            TemporalMetadata::Timestamp(TimeUnit::D, _) => {
+                vortex_bail!(InvalidArgument: "Invalid TimeUnit TimeUnit::D for TemporalMetadata::Timestamp")
+            }
+            TemporalMetadata::Timestamp(unit, None) => Ok(TemporalJiff::Timestamp(
+                Timestamp::UNIX_EPOCH.checked_add(unit.to_jiff_span(v))?,
+            )),
+            TemporalMetadata::Timestamp(unit, Some(tz)) => Ok(TemporalJiff::Zoned(
+                Timestamp::UNIX_EPOCH
+                    .checked_add(unit.to_jiff_span(v))?
+                    .intz(tz)?,
```

Comment:
miri doesn't like this call because it reads file from the os. We should ignore tests using this with miri using `#[cfg_attr(miri, ignore)]`